pkg/apis/monitoring/v1/types.go

```diff
@@ -1816,22 +1816,34 @@ type PrometheusRuleSpec struct {
 // or 'abort'.  More info: https://github.com/thanos-io/thanos/blob/main/docs/components/rule.md#partial-response
 // +k8s:openapi-gen=true
 type RuleGroup struct {
-	Name                    string `json:"name"`
-	Interval                string `json:"interval,omitempty"`
-	Rules                   []Rule `json:"rules"`
+	// Name of the rule group.
+	Name string `json:"name"`
+	// Interval determines how often rules in the group are evaluated.
+	Interval string `json:"interval,omitempty"`
+	// Alerting or Recording Rules
+	Rules []Rule `json:"rules"`
+	// PartialResponseStrategy is only used by ThanosRuler
+	// +kubebuilder:validation:Enum="";ABORT;WARN
+	// +kubebuilder:default:=""
 	PartialResponseStrategy string `json:"partial_response_strategy,omitempty"`
 }
 
 // Rule describes an alerting or recording rule
 // See Prometheus documentation: [alerting](https://www.prometheus.io/docs/prometheus/latest/configuration/alerting_rules/) or [recording](https://www.prometheus.io/docs/prometheus/latest/configuration/recording_rules/#recording-rules) rule
 // +k8s:openapi-gen=true
 type Rule struct {
-	Record      string             `json:"record,omitempty"`
-	Alert       string             `json:"alert,omitempty"`
-	Expr        intstr.IntOrString `json:"expr"`
-	For         string             `json:"for,omitempty"`
-	Labels      map[string]string  `json:"labels,omitempty"`
-	Annotations map[string]string  `json:"annotations,omitempty"`
+	// Name of the time series to output to. Must be a valid metric name.
+	Record string `json:"record,omitempty"`
+	// Name of the alert. Must be a valid label value.
+	Alert string `json:"alert,omitempty"`
+	// PromQL expression to evaluate.
+	Expr intstr.IntOrString `json:"expr"`
+	// Alerts are considered firing once they have been returned for this long.
+	For string `json:"for,omitempty"`
+	// Labels to add or overwrite for each alert.
```

Comment:
Applies for both alerting and recording rule.

```suggestion
	// Labels to add or overwrite.
```